test/interface/ad_tests.jl

```diff
@@ -208,3 +208,61 @@ end
 
 @test !iszero(ForwardDiff.gradient(t -> of_a(t), [1.0]))
 @test ForwardDiff.gradient(t -> of_a(t), [1.0]) ≈ FiniteDiff.finite_difference_gradient(t -> of_a(t), [1.0]) rtol=1e-5
+
+
+SOLVERS_FOR_AD = (
+    BS3,
+    Tsit5,
+    KenCarp4, KenCarp47, KenCarp5, KenCarp58,
+    TRBDF2,
+    RadauIIA3, RadauIIA5,
+    Rodas4, Rodas5,
+    Rosenbrock23, Rosenbrock32,
+    Vern6, Vern7,
+    )
+
+@testset "$alg can handle ForwardDiff.Dual in u0 when iip=$iip" for
+    alg in SOLVERS_FOR_AD,
+    iip in (true, false)
+
+    if iip
+      f = (du, u, p, t) -> du .= -0.5*u
+    else
+      f = (u, p, t) -> -0.5*u
+    end
+
+    g = u0 -> begin
+        tspan = (0.0, 1.0)
+        prob = ODEProblem(
+            f,
+            u0,
+            tspan
+        )
+        solve(prob, alg())(last(tspan))[1]

```

Comment:
can you set the solver tolerance to `abstol=1e-14,reltol=1e-14`, and lower the test tolerance to something to match? With analytical solution, that test could probably be to like `1e-12`.

Reply:
I've not tightened the tolerances in the tests. I've set the test tolerance as tight as possible for each of the algorithms. Notice that I have to use a pretty large relative error for `TRBDF2`.